raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -306,10 +314,22 @@ private VoteResponseData buildVoteResponse(Errors error, boolean voteGranted) {
                 .setVoteGranted(voteGranted);
     }
 
-    private VoteResponseData handleVoteRequest(VoteRequestData request) throws IOException {
-        Optional<Exception> errorOpt = validateLeaderOnlyRequest(request.candidateId(), request.candidateEpoch());
+    /**
```

Comment:
It's a little annoying to duplicate it since that it's hard to keep them in sync, so I just decided to document the expected errors on the request handler.